.github/workflows/relay.yml

```diff
@@ -63,3 +63,90 @@ jobs:
             /tmp/gotest.log
             ./race_coverage.txt
             ./integration_coverage.txt
+
+  get_projectserum_version:
+    name: Get ProjectSerum Version
+    runs-on: ubuntu-latest
+    outputs:
+      projectserum_version: ${{ steps.psversion.outputs.projectserum_version }}
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+      - name: Get ProjectSerum Version
+        id: psversion
+        uses: ./.github/actions/projectserum_version
+
+  build_wrapped_anchor_image:
+    name: build contract test image
+    runs-on: ubuntu-latest
+    needs: [get_projectserum_version]
+    defaults:
+      run:
+        working-directory: contracts
+    steps:
+    - uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+    - name: cache docker build image
+      id: cache-image
+      uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
+      with:
+        path: contracts/docker-build.tar
+        key: ${{ runner.os }}-docker-pnpm-build-${{ needs.get_projectserum_version.outputs.projectserum_version }}-${{ hashFiles('**/Cargo.lock') }}
+    - name: build & save image
+      if: steps.cache-image.outputs.cache-hit != 'true'
+      run: |
+        docker buildx build . -t chainlink-solana:build --build-arg ANCHOR_CLI=${{ needs.get_projectserum_version.outputs.projectserum_version }}
+        docker save chainlink-solana > docker-build.tar
```

Comment:
I tried pulling from the cache without building the image, but that didn't work since the cache wasn't in the workflow. I can move this test to the contract tests for now to reduce build time and resource usage.